NMS-15425: Update dual write plugin docs (#5788)
OpenNMS/opennms: docs/modules/deployment/pages/time-series-storage/timeseries/time-series-storage.adoc

== Run two time series writes in parallel

The default time series strategy (RRD, Newts, Integration Plugin) is used for both read and write operations.
You can add another time series to write in parallel to the default strategy.
This additional time series database may be used for validation purposes or for staging metrics before moving from one time series to another.
If you plan to migrate to a new storage strategy, you can set the new strategy as a write destination while keeping reads on your current storage.
Once you have written enough data to the new storage, you can then flip the reads over to the new system.

[[ga-dual-write-newts]]
=== Write to Newts in parallel

> You can enable writing to Newts without it being the default time series strategy by configuring Newts but not setting it was the `org.opennms.timeseries.strategy`.

Reviewer: Check the wording here.

== Run two time series writes in parallel

The default time series strategy (RRD, Newts, Integration Plugin) is used for both read and write operations.
You can add another time series to write in parallel to the default strategy.
This additional time series database may be used for validation purposes or for staging metrics before moving from one time series to another.
If you plan to migrate to a new storage strategy, you can set the new strategy as a write destination while keeping reads on your current storage.
Once you have written enough data to the new storage, you can then flip the reads over to the new system.

> In order for the dual write plugin to work, you must install a time series feature (either Newts or an integration plugin) and set `org.opennms.timeseries.strategy` to `rrd`.

Author: ```suggestion
In order for the dual write plugin to work, you must install a time series feature (either Newts or an integration plugin) and leave `org.opennms.timeseries.strategy` at your current value.
```

[source, console]
sudo systemctl restart opennms

[[ga-dual-write-integration]]
=== Write to Time Series Integration Plugin in parallel

You can enable writing to any time series integration plugin in parallel without having integration as the default time series strategy by enabling the `opennms-timeseries-api` feature.

> This will direct writes to both your `org.opennms.timeseries.strategy` destination and any installed timeseries integration plugin.

Reviewer: We need to clarify this, I think:
> This will direct writes to both your `org.opennms.timeseries.strategy` destination and any installed timeseries integration plugin.

We need to be clear in the dual-write scenario, `org.opennms.timeseries.strategy` must be set to `rrd` (or not set at all, in which case the default is `rrd`) the dual-writes to the integration strategy to work. When `opennms-timeseries-api` is loaded, the secondary write to the integration strategy happens implicitly.

[source, console]
sudo systemctl restart opennms

[[ga-dual-write-integration]]

> === Write to Time Series Integration Plugin in parallel

Reviewer 2: ```suggestion
=== Write to time series integration plugin in parallel
```

> You can enable writing to any Time Series Integration plugin in parallel to RRD or Newts by enabling the `opennms-timeseries-api` feature and not setting the strategy to `integration`.

Reviewer 2: ```suggestion
You can enable writing to any time series integration plugin in parallel to RRD or Newts by enabling the `opennms-timeseries-api` feature and not setting the strategy to `integration`.
```

> You can enable writing to any time series integration plugin in parallel to RRD or Newts by enabling the `opennms-timeseries-api` feature and not setting the strategy to `integration`.

Reviewer: In one other spot, we suggest setting it to `rrd`. We might want to be consistent. https://github.com/OpenNMS/opennms/pull/5788/files#diff-d7bbf1320c17520776181e8ec0528061d849967d21c1a02c826bb90a92203910R54

Author: see comment at linked line.

> === Write to Time Series Integration Plugin in parallel
> 
> You can enable writing to any time series integration plugin in parallel without having integration as the default time series strategy by enabling the `opennms-timeseries-api` feature.

Author: Any TSIL plugin.
```suggestion
You can enable writing to any Time Series Integration plugin in parallel to RRD or Newts by enabling the `opennms-timeseries-api` feature and not setting the strategy to `integration`.
```
